src/sinks/nats.rs

```diff
@@ -157,14 +181,18 @@ impl StreamSink<Event> for NatsSink {
                 }
             };
 
-            let log = encode_event(event, &self.encoding);
-            let message_len = log.len();
+            self.transformer.transform(&mut event);
 
-            match self.connection.publish(&subject, log).await {
+            let mut bytes = BytesMut::new();
+            if self.encoder.encode(event, &mut bytes).is_err() {
+                continue;
+            }
```

Comment:
`Encoder` itself will always emit an `EncoderSerializeFailed`/`EncoderFramingFailed` internal event:

https://github.com/vectordotdev/vector/blob/9658477bf36d7972a5da1e861730bf7701997911/src/codecs/encoder.rs#L56-L62

https://github.com/vectordotdev/vector/blob/9658477bf36d7972a5da1e861730bf7701997911/src/codecs/encoder.rs#L131-L135

Reply:
Ah, gotcha, didn't realize it already emitted logs. Might be worth throwing a comment here since I can see myself stumbling upon this in 6 months and having the same question 😅 .

Reply:
Added in 84cc9ef90cdac105e4dd093fa125bbc86fd65f38

Reply:
That does kinda beg the question, if the error are already handled then should we also be returning them?

Reply:
Yeah, I had a similar thought but forgot to leave it. Maybe `encode` should just return an `Option`?

Reply:
Hm, yeah it would be possible to provide another method with that signature. That `encode` here specifically comes from the `tokio_util::codec::Encoder` trait, though